EngineLayer/Proteomics/CompactPeptide.cs

```diff
@@ -44,6 +44,13 @@ public CompactPeptide(PeptideWithSetModifications peptideWithSetModifications)
             MonoisotopicMassIncludingFixedMods = peptideWithSetModifications.MonoisotopicMass;
         }
 
+        public CompactPeptide(CompactPeptide cp, double MonoisotopicMassIncludingFixedMods)
+        {
+            this.CTerminalMasses = cp.CTerminalMasses;
+            this.NTerminalMasses = cp.NTerminalMasses;
+            this.MonoisotopicMassIncludingFixedMods = MonoisotopicMassIncludingFixedMods;
+        }
```

Comment:
What's the purpose of this constructor? It doesn't quite look like a copy-constructor, since the CTerminalMasses arrays in both CompactPeptides reference the same array